youtube_dl/extractor/canvas.py

```diff
@@ -110,10 +104,153 @@ def _real_extract(self, url):
+                'duration': 3011.17,
+                'thumbnail': r're:^https?://.*\.jpg$'
+            },
+            'skip': 'This video is only available for registered users'
+        }
+    ]
+    _NETRC_MACHINE = 'vrtnu'
+    _APIKEY = '3_0Z2HujMtiWq_pkAjgnS2Md2E11a1AwZjYiBETtwNE-EoEHDINgtnvcAOpNgmrVGy'
+    _CONTEXT_ID = 'R1070628488'
+
+    def _real_initialize(self):
+        self._login()
+
+    def _login(self):
+        username, password = self._get_login_info()
+        if username is None:
+            self.raise_login_required()
+
+        auth_data = {
+            'APIKey': self._APIKEY,
+            'targetEnv': 'jssdk',
+            'loginID': username,
+            'password': password,
+            'authMode': 'cookie',
+        }
+
+        auth_info = self._download_json(
+            'https://accounts.eu1.gigya.com/accounts.login', None,
+            note='Logging in', errnote='Unable to log in',
+            data=urlencode_postdata(auth_data))
+
+        error_message = auth_info.get('errorDetails') or auth_info.get('errorMessage')
+        if error_message:
+            raise ExtractorError(
+                'Unable to login: %s' % error_message, expected=True)
+
+        # When requesting a token, no actual token is returned, but the
+        # necessary cookies are set.
+        self._request_webpage(
+            'https://token.vrt.be',
+            None, note='Requesting a token', errnote='Could not get a token',
+            headers={
+                'Content-Type': 'application/json',
+                'Referer': 'https://www.vrt.be/vrtnu/',
+            },
+            data=json.dumps({
+                'uid': auth_info['UID'],
+                'uidsig': auth_info['UIDSignature'],
+                'ts': auth_info['signatureTimestamp'],
+                'email': auth_info['profile']['email'],
+            }).encode('utf-8'))
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        display_id = mobj.group('id')
+
+        webpage = self._download_webpage(url, display_id)
+
+        title = self._html_search_regex(
+            r'<h1 class="content__heading">(.+?)</h1>',
+            webpage, 'title', default=None, flags=(re.M | re.S)).strip()
+
+        description = self._html_search_regex(
+            r'<div class="content__description">(.+?)</div>',
+            webpage, 'description', default=None, flags=(re.M | re.S))
+
+        season = self._html_search_regex(
+            [r'''(?xms)<div\ class="tabs__tab\ tabs__tab--active">\s*
+                    <span>seizoen\ (.+?)</span>\s*
+                </div>''',
+             r'<option value="seizoen (.+?)" data-href="[^"]+?" selected>'],
+            webpage, 'season', default=None)
+
+        season_number = int_or_none(season)
+
+        # Sometimes the season is the year, so don't use it as the season number
+        if season_number and season_number > 1000:
```

Comment:
If I do this with a regex and the season is something like for example "2017", both `season` and `season_number` will be `None`. Whereas previously, `season` would still be `"2017"`. For me the new behaviour is fine, as I don't care about non-numbered seasons.